src/ecdsa/ellipticcurve.py

```diff
@@ -574,6 +584,7 @@ def __mul__(self, other):
         if self.__order:
             # order*2 as a protection for Minerva
             other = other % (self.__order * 2)
+        self._maybe_precompute()
```

Comment:
this will escalate the lock from reader to writer! not sure ... is "getstate" supposed to return also the lazy-computed attribute?

Reply:
> this will escalate the lock from reader to writer! not sure ...

I don't see how...

> is "getstate" supposed to return also the lazy-computed attribute?

yes, it should; the pickling is used for multiprocess module for example